src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitionInfo.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Syntax.SourceInformation;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    internal class UserDefinitionInfo

```

Comment:
consider changing this class name to be clearer that it's for maintaining source info/trivia for prettyprint and similar operations.